src/__tests__/core/index.test.js

```diff
@@ -47,19 +49,199 @@ describe('setup', () => {
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://my-tenant.us.auth0.com');
+    });
+
+    it('should use the clientBaseUrl option if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          clientBaseUrl: 'https://client-base-url.example.com',
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://client-base-url.example.com');
+    });
+
+    it('should use configurationBaseUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          configurationBaseUrl: 'https://config-base-url.example.com',
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://config-base-url.example.com');
+    });
+
+    it('should use assetsUrl if given', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          assetsUrl: 'https://assets-url.example.com'
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.clientBaseUrl).toBe('https://assets-url.example.com');
+    });
+  });
+
+  describe('tenantBaseUrl', () => {
+    it('tenantBaseUrl should default to domain URL when using auth0.com', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'my-tenant.us.auth0.com',
+        {
+          __useTenantInfo: true
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.tenantBaseUrl).toBe('https://my-tenant.us.auth0.com/tenants/v1/my-tenant.js');
+    });
+
+    it('should default to domain URL when using a custom domain', () => {
+      const { mock } = mockInit;
+
+      setup(
+        'id',
+        'clientID',
+        'auth.my-tenant.com',
+        {
+          __useTenantInfo: true
+        },
+        'hookRunner',
+        'emitEventFn',
+        'handleEventFn'
+      );
+
+      expect(mock.calls.length).toBe(1);
+
+      const model = mock.calls[0][1].toJS();
+      expect(model.tenantBaseUrl).toBe('https://auth.my-tenant.com/info-v1.js');
```

Comment:
why the path differs in this one? what's the condition? you might want to reflect that in the test name

Reply:
Yeah I'm not sure of the history as to why this is. If you use a custom domain, the URL differs when using `__useTenantInfo` vs using an `auth0.com` domain. Just trying to keep and verify the same logic here.